p2p/client.go

```diff
@@ -143,8 +143,12 @@ func (s *peer) getSignatureForDocument(ctx context.Context, cd coredocumentpb.Co
 		if err != nil {
 			return nil, err
 		}
-
-		resp, err = h.RequestDocumentSignature(localPeerCtx, &p2ppb.SignatureRequest{Document: &cd})
+		id, err := tc.GetIdentityID()
```

Comment:
this is not required since the `tc` is derived from `id` in the top. You can directly pass it as is